pkg/pillar/cmd/volumemgr/handlework.go

```diff
@@ -134,155 +49,57 @@ type casIngestWorkResult struct {
 	loaded            []string
 }
 
-// Map with the result
-// Caller needs to do a delete after the lookup
-var volumeWorkResultMap = make(map[string]volumeWorkResult)
-var casIngestWorkResultMap = make(map[string]casIngestWorkResult)
-
-func lookupVolumeWorkResult(ctx *volumemgrContext, key string) *volumeWorkResult {
-	res, ok := volumeWorkResultMap[key]
-	if ok {
-		return &res
-	} else {
-		return nil
-	}
-}
-
-func addVolumeWorkResult(ctx *volumemgrContext, key string, res volumeWorkResult) {
-	volumeWorkResultMap[key] = res
-}
-
-func deleteVolumeWorkResult(ctx *volumemgrContext, key string) {
-	delete(volumeWorkResultMap, key)
-}
-
-func lookupCasIngestWorkResult(ctx *volumemgrContext, key string) *casIngestWorkResult {
-	if res, ok := casIngestWorkResultMap[key]; ok {
-		return &res
-	}
-	return nil
-}
-
-func addCasIngestWorkResult(ctx *volumemgrContext, key string, res casIngestWorkResult) {
-	casIngestWorkResultMap[key] = res
-}
-
-func deleteCasIngestWorkResult(ctx *volumemgrContext, key string) {
-	delete(casIngestWorkResultMap, key)
-}
-
-// MaybeAddWorkCreate checks if the Key is in the map of pending work
+// AddWorkCreate checks if the Key is in the map of pending work
 // and if not kicks of a worker and adds it
 // XXX defer if busy?
-func MaybeAddWorkCreate(ctx *volumemgrContext, status *types.VolumeStatus) {
-	log.Infof("MaybeAddWorkCreate(%s)", status.Key())
-	if lookupPendingCreate(ctx, status.Key()) {
-		log.Infof("MaybeAddWorkCreate(%s) found", status.Key())
-		return
-	}
+func AddWorkCreate(ctx *volumemgrContext, status *types.VolumeStatus) {
 	d := volumeWorkDescription{
 		create: true,
 		status: *status,
 	}
-	w := worker.Work{Key: status.Key(), Description: d}
+	w := worker.Work{Kind: workCreate, Key: status.Key(), Description: d}
 	// XXX could check a return and not add...
 	ctx.worker.Submit(w)
-	// XXX success - add
-	addPendingCreate(ctx, status.Key())
-	log.Infof("MaybeAddWorkCreate(%s) done", status.Key())
 }
 
-// MaybeAddWorkLoad checks if the Key is in the map of pending work
+// AddWorkLoad checks if the Key is in the map of pending work
```

Comment:
This comment has gone stale; there is no longer a map of pending work.

Reply:
Done